app/models/account.rb

```diff
@@ -110,7 +110,12 @@ def self.columns
   before_destroy :destroy_features
 
   scope :free, ->(free_date) { where.has { not_exists Contract.have_paid_on(free_date).by_account(BabySqueel[:accounts].id).select(:id) } }
-  scope :not_enterprise, -> { where.has { not_exists Service.of_account(BabySqueel[:accounts].id).with_enterprise_application_plans.select(:id) } }
+
+  scope :without_bought_application_plans_with_system_names, lambda { |system_names|
```

Comment:
Why can't I think of a better name to simplify this scope? Any other idea?

Reply:
my only idea is changing `bought_application_plans` for `cinstances` or `applications` so it is a bit shorter 😂